pkg/storage/wal/manager.go

```diff
@@ -81,6 +83,29 @@ type Config struct {
 	MaxSegmentSize int64
 }
 
+type Metrics struct {
+	NumAvailable prometheus.Gauge
+	NumPending   prometheus.Gauge
+	NumFlushing  prometheus.Gauge
+}
+
+func NewMetrics(r prometheus.Registerer) *Metrics {
+	return &Metrics{
+		NumAvailable: promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Name: "wal_segments_available",
```

Comment:
I think we need a prefix here, but since `Namespace` and `Subsystem` are deprecated, I'm not sure if I should hardcode it or accept it as an argument to `NewMetrics`.